src/Features/CSharp/Portable/RemoveUnusedParametersAndValues/CSharpRemoveUnusedParametersAndValuesCodeFixProvider.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.RemoveUnusedParametersAndValues;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnusedParametersAndValues
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.RemoveUnusedParametersAndValues), Shared]
+    internal class CSharpRemoveUnusedParametersAndValuesCodeFixProvider:
+        AbstractRemoveUnusedParametersAndValuesCodeFixProvider<ExpressionSyntax, StatementSyntax, BlockSyntax, 
+                                                               ExpressionStatementSyntax, LocalDeclarationStatementSyntax,
+                                                               VariableDeclaratorSyntax, ForEachStatementSyntax,
+                                                               SwitchSectionSyntax, SwitchLabelSyntax,
+                                                               CatchClauseSyntax, CatchClauseSyntax>
+    {
+        protected override BlockSyntax GenerateBlock(IEnumerable<StatementSyntax> statements)
+            => SyntaxFactory.Block(statements);
+
+        protected override SyntaxToken GetForEachStatementIdentifier(ForEachStatementSyntax node)
+            => node.Identifier;
+
+        protected override SyntaxNode UpdateNameForFlaggedNode(SyntaxNode node, SyntaxToken newName)
```

Comment:
definitely curious what this is needed for.

Reply:
Added comment.

---
In reply to: [230603185](https://github.com/dotnet/roslyn/pull/30777#discussion_r230603185) [](ancestors = 230603185)